components/br-stream/src/endpoint.rs

```diff
@@ -0,0 +1,184 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::fmt;
+use tokio::io::Result as TokioResult;
+use tokio::runtime::Runtime;
+use tokio_stream::StreamExt;
+
+use crate::metadata::store::MetaStore;
+use crate::metadata::{MetadataClient, MetadataEvent, Task as MetaTask};
+use crate::{errors::Result, observer::BackupStreamObserver};
+use online_config::ConfigChange;
+use raftstore::coprocessor::CmdBatch;
+use tikv::config::BackupConfig;
+use tikv_util::worker::{Runnable, Scheduler};
+use tikv_util::{debug, error, info};
+
+pub struct Endpoint<S: MetaStore + 'static> {
+    config: BackupConfig,
+    meta_client: MetadataClient<S>,
+    scheduler: Scheduler<Task>,
+    observer: BackupStreamObserver,
+    pool: Runtime,
+}
+
+impl<S> Endpoint<S>
+where
+    S: MetaStore + 'static,
+{
+    pub fn new(
+        meta_client: MetadataClient<S>,
+        config: BackupConfig,
+        scheduler: Scheduler<Task>,
+        observer: BackupStreamObserver,
+    ) -> Endpoint<S> {
+        let pool = create_tokio_runtime(config.num_threads, "br-stream")
```

Comment:
`config.num_threads` is 75% of vcpu by default, maybe too huge?

Reply:
OK, may be we should use an independent config like `BackupStreamConfig`. to avoid this.